Add option to not mount home directory

jmstover opened the pull request:
**Description of the Pull Request (PR):**

This PR adds a `--no-home` option that will *not* mount the users home directory even if `mount home = yes` is set in `singularity.conf`

This will still mount your $HOME if it is also your current working directory. There is not logic to circumvent that. So, normal usage would be something like:

```
$ cd /tmp
$ singularity exec --no-home ~/ubuntu.img ls -l $HOME
ls: cannot access '/home/jason': No such file or directory
```

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [x] I have tested this PR locally with a `make test`
- [x] This PR is NOT against the project's master branch
- [] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

jmstover committed:
Add option to not mount user home

jmstover committed:
Move NOHOME check higher in chain.

Also, check against a NULL value from singularity_registry_get()

jmstover committed:
Add test for --no-home

jmstover committed:
Add help text

GodloveD reviewed (dismissed):
Looks sensible (though I haven't tested yet).  Can you add entry to changelog and bash completion script please?  (You knew I was going to ask!)

jmstover committed:
Requested updates

  - Add entry to CHANGELOG.md
  - Add `--no-home` option to bash completion.

cclerget reviewed (approved):
Looks good :+1:

jmstover committed:
Force pylint version

pylint was recently updated, and "seems" to have broken things. Try
the old version.

GodloveD reviewed (approved):
lgtm, and works with my tests.
